src/runtime/action/index.ts

```diff
@@ -0,0 +1,8 @@
+export interface ActionReturn<Parameters = any> {
+	update?: (parameters: Parameters) => void;
+	destroy?: () => void;
+}
+
+export interface Action<Parameters = any, Element = HTMLElement> {
+	<Node extends Element>(node: Node, parameters?: Parameters): void | ActionReturn<Parameters>;
```

Comment:
Re `parameters?: Parameters`, do we want to remove the `?` so someone could make the type stricter? Also since in `ActionReturn` type, the `update` parameters is non-optional too.

Reply:
Good catch! We actually want to make this optional because `use:action` without parameters is valid per docs and parser, and making it required would force users to supply an argument, `use:action={{}}`. Without `?`, TS will complain that it expects 2 arguments but only 1 with the former.

They can actually work around this with `Action<void>`, but rather than doing so, we'll leave it to the authors to supply a default value for their action parameters, or have a runtime check as well to make sure an argument is supplied.

Reply:
Although, as soon as I comment this, I realize we can indeed remove the optional mark and have the default value for parameters as `void`. This gives the same type-safety and better convenience for TS users, but as always, it wouldn't have any effect for JS-only users, while the current definition would require authors to implement runtime guards that would, as a side effect, also affects JS-only users.

Reply:
I thought with parameters as `any`, it would also cover `undefined` too, so `use:action` without parameters would still pass. Or maybe my TS-foo tricked me. But your second comment sounds like it could work too.